eo-maven-plugin/src/main/java/org/eolang/maven/objectionary/ObjsDefault.java

```diff
@@ -23,43 +23,144 @@
  */
 package org.eolang.maven.objectionary;
 
+import java.io.IOException;
+import java.nio.file.Path;
 import java.util.HashMap;
 import java.util.Map;
+import org.cactoos.Input;
+import org.cactoos.Scalar;
+import org.cactoos.iterable.Mapped;
+import org.cactoos.map.MapEntry;
+import org.cactoos.map.MapOf;
+import org.cactoos.scalar.Unchecked;
+import org.eolang.maven.hash.ChCached;
+import org.eolang.maven.hash.ChNarrow;
 import org.eolang.maven.hash.CommitHash;
 
 /**
  * Default objectionaries.
+ * The class is immutable, but NOT thread-safe.
  * @since 0.29.6
  */
 public final class ObjsDefault implements Objectionaries {
+
+    /**
+     * Cache path.
+     */
+    private final Unchecked<Path> cache;
+
+    /**
+     * Use cache.
+     */
+    private final Scalar<Boolean> usehash;
+
     /**
      * Hash-map.
      */
-    private final Map<String, Objectionary> map;
+    private final Map<? super String, Objectionary> map;
 
     /**
-     * Ctor.
+     * Constructor.
+     * @param cache Cache path.
+     * @param usehash Use cache.
      */
-    public ObjsDefault() {
-        this(new HashMap<>());
+    public ObjsDefault(final Scalar<Path> cache, final Scalar<Boolean> usehash) {
+        this(cache, usehash, new HashMap<>(0));
     }
 
     /**
-     * Ctor.
-     * @param ojs Objectionaries hash-map.
+     * Constructor for tests with predefined Objectionaries.
+     * @param entries Predefined Objectionaries.
      */
-    ObjsDefault(final Map<String, Objectionary> ojs) {
-        this.map = ojs;
+    @SafeVarargs
+    public ObjsDefault(final Map.Entry<CommitHash, Objectionary>... entries) {
+        this(new Mapped<>(e -> new MapEntry<>(e.getKey().value(), e.getValue()), entries));
```

Comment:
@volodya-lombrozo constructor accepts `entries` where every entry is `MapEntry`. You "map" this `entries` and make from it `Iterable` of `MapEntry` again.
So you take every entry, take key and value from it and create the same entry...
Here you can do just `new MapOf<>(entries)` (it will work, I've just tested it) and after that you won't need private constructor above that accepts `Iterable`.

Reply:
@maxonfjvipon Please, take a close look again :)

1. Pay attention to types in diamond brackets 
2. `So you take every entry, take key and value from it and create the same entry...` -> it is not true.
3. As for `new MapOf<>(entries)` You tested only compilation I guess? Try this one: `new MapOf<String, Objectionary>(entries)`. This [thread](https://stackoverflow.com/questions/12462079/possible-heap-pollution-via-varargs-parameter#:~:text=Heap%20pollution%20is%20a%20technical,the%20object%20they%20point%20to.) could help.

Reply:
@volodya-lombrozo sorry, my bad, now I can see... keys are translated from `CommitHash` to `String`. No more claims here